pkg/infrastructure/services/deployer.go

```diff
@@ -295,3 +310,39 @@ func (s *serviceDeployer) update() error {
 	}
 	return nil
 }
+
+// checkInfraDependencies verifies if every KogitoInfra resource have an ok status.
+func (s *serviceDeployer) checkInfraDependencies() (time.Duration, error) {
+	kogitoInfraReferences := s.instance.GetSpec().GetInfra()
+	log.Debugf("Going to fetch kogito infra properties for given references : %s", kogitoInfraReferences)
+	for _, infraName := range kogitoInfraReferences {
+		infra, err := infrastructure.FetchKogitoInfraInstance(s.client, infraName, s.instance.GetNamespace())
+		if err != nil {
+			return reconciliationPeriodAfterInfraError, err
```

Comment:
Actually, no. The `v1alpha1.FailureInfraConditionType` defines lots of status now, which can be viewed in the kogitoinfra and kogitoruntime `status` description. Returning a reconciliation period instead of an error will make the less reconciliation loops. But this function can be better: we can return the `exists` flag and check if it's not valid passing a more meaningful message to the user. I'll do it.